storage-proofs/post/src/fallback/compound.rs

```diff
@@ -68,7 +68,11 @@ impl<'a, Tree: 'static + MerkleTreeTrait>
                     &pub_params,
                     pub_inputs.randomness,
                     sector.id.into(),
-                    challenge_index as u64,
+                    if pub_params.is_winning {
+                        i
+                    } else {
+                        challenge_index
+                    } as u64,
```

Comment:
This works, but it's a hack which now depends on each Winning PoSt sector being challenged only once. To be more correct, the `true` branch of this conditional should fully restore previous behavior. In practice, this won't matter with current parameterization, but we shouldn't depend on that. Certainly we would need at least an assertion if we intended to. That might actually be preferable. We have an enormous amount of complexity in place to support a generality I don't think we need. If we can close off some possibilities for the sake of simpler, clearer, code that might be worth doing.